src/Bicep.Core/Emit/TemplateEmitter.cs

```diff
@@ -29,30 +33,50 @@ public TemplateEmitter(SemanticModel model)
     /// </summary>
     /// <param name="stream">The stream to write the template</param>
     /// <param name="existingContent">Existing content of the parameters file</param>
-    public EmitResult EmitEmptyParametersFile(Stream stream, string existingContent)
+    /// <param name="outputFormat">Output file format (json or bicepparam)</param>
+    /// <param name="includeParams">Include parameters (requiredonly or all)</param>
+    public EmitResult EmitEmptyParametersFile(Stream stream, string existingContent, OutputFormatOption outputFormat, IncludeParamsOption includeParams)
     {
         using var sw = new StreamWriter(stream, UTF8EncodingWithoutBom, 4096, leaveOpen: true);
 
-        return EmitEmptyParametersFile(sw, existingContent);
+        return EmitEmptyParametersFile(sw, existingContent, outputFormat, includeParams);
     }
 
     /// <summary>
     /// Emits a template to the specified json writer if there are no errors. No writes are made to the writer if there are compilation errors.
     /// </summary>
     /// <param name="textWriter">The text writer to write the template</param>
     /// <param name="existingContent">Existing content of the parameters file</param>
-    public EmitResult EmitEmptyParametersFile(TextWriter textWriter, string existingContent) => this.EmitOrFail(() =>
+    /// <param name="outputFormat">Output file format (json or bicepparam)</param>
+    /// <param name="includeParams">Include parameters (requiredonly or all)</param>
+    public EmitResult EmitEmptyParametersFile(TextWriter textWriter, string existingContent, OutputFormatOption outputFormat, IncludeParamsOption includeParams) => this.EmitOrFail(() =>
```

Comment:
is `EmitEmpty` in the function name here implying that we are emitting a parameter file with no values for parameters? If so then maybe we should think about renaming it? `EmitTemplateGeneratedParameterFile`?

Reply:
when I first introduced this function, I was following an example on another file. I can't find that sample anymore, probably it's renamed along the way.

so, I renamed this function as well.

thanks for the name suggestion 👍